### PR TITLE
US2.4 Generate and print board

### PR DESCRIPTION
**Pull request by OriaTori**
#19

### CODE REVIEW

File: src/DisplayBoard.cpp [hunk 1]
```diff
@@ -51,7 +51,7 @@ std::string display(const Board& b)
             output << drawRow(widthRows, {});
         else
             output << drawRow(widthRows, cluesRows.at(i));
-        output << drawEmptyLine(width, '?') << "\n";
+        output << drawBoardLine(width, const_cast<Board&>(b).getBoardLines().at(i)) << "\n";
```
Comment (LordLukin) [inline src/DisplayBoard.cpp]: Probably when you pass a `BLine const &` then you will not need a `const_cast` here. But I may be wrong. Compiler will tell you the truth :)

File: test/DisplayBoardTests.cpp [hunk 2]
```diff
@@ -59,3 +59,20 @@ TEST_F(DisplayBoardTests, checkDisplayBoardWithClues)
     compareBoard = "     +------+\n     |     1|\n     |     1|\n     | 1   1|\n+----+------+\n|   1|??????|\n|   1|??????|\n| 1 1|??????|\n+----+------+\n";
     ASSERT_EQ(display(board5), compareBoard);
 }
+
+TEST_F(DisplayBoardTests, checkDisplayBoardWithDifferentBoardFields)
+{
+    Board board(2, 2, {{1},{1}}, {{1},{1}});
+    std::string compareBoard = "   +----+\n   | 1 1|\n+--+----+\n| 1|????|\n| 1|????|\n+--+----+\n";
```
Comment (LordLukin) [inline test/DisplayBoardTests.cpp]: expectedBoard will be a better name

File: src/DisplayBoard.cpp (hunk 1 above)
Comment (OriaTori) [reply src/DisplayBoard.cpp]: getBoardLines() return  BLines& - and unfortunately do not return a const object, that is why I use in this line const_cast.
 In other tasks, to solving algorithm there will be used getBoardLines() to get value and changed it. 
I really don't know how in another way get board_ as a BLine & const - and I don't think that writing the second method will be a good solution...so, I don't know.

Comment (LordLukin) [reply src/DisplayBoard.cpp]: Take a look at my changes: https://github.com/LordLukin/LogicalImages/pull/51/commits/6f3d0aba23ae3b5e031ded28843e2c41b70d4c6b
There were some missing consts in `getBoardLines()` function. Now it works without const_cast :)

File: test/DisplayBoardTests.cpp (hunk 2 above)
Comment (OriaTori) [reply test/DisplayBoardTests.cpp]: done

**Pull request merged**